src/rdkafka_broker.c

```diff
@@ -3687,39 +3757,55 @@ rd_kafka_fetch_reply_handle (rd_kafka_broker_t *rkb,
                                 int16_t ErrorCode;
                                 int64_t HighwaterMarkOffset;
                                 int64_t LastStableOffset;       /* v4 */
+                                int64_t LogStartOffset;         /* v5 */
                                 int32_t MessageSetSize;
+                                int32_t PreferredReadReplica;   /* v11 */
                         } hdr;
                         rd_kafka_resp_err_t err;
                         int64_t end_offset;
 
 			rd_kafka_buf_read_i32(rkbuf, &hdr.Partition);
 			rd_kafka_buf_read_i16(rkbuf, &hdr.ErrorCode);
+                        if (ErrorCode)
+                                hdr.ErrorCode = ErrorCode;
 			rd_kafka_buf_read_i64(rkbuf, &hdr.HighwaterMarkOffset);
 
                         end_offset = hdr.HighwaterMarkOffset;
 
-                        if (rd_kafka_buf_ApiVersion(request) == 4) {
+                        hdr.LastStableOffset = -1;
```

Comment:
yeah totally agree. this will change what's reported in stats from the previous release which is technically breaking, but not a big deal. Also appropriate for LogStartOffset, though that has no consequence.